### PR TITLE
Aa/demo forward tacotron attention

### PR DESCRIPTION
**Pull request by a-a-egorovich**
Demo for modified ForwardTacotron and MelGAN pipeline.

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins ok to test

File: demos/text_to_speech_demo/python/models/mel2wave_ie.py [hunk 1]
```diff
@@ -178,3 +178,95 @@ def forward_rnn(self, mels, upsampled_mels, aux):
...
+        """
+        return class provided MelGAN inference.
+
+        :param model: path to xml with MelGAN model of WaveRNN
+        :param ie: instance of the IECore
+        :param device: target device
+        :return:
+        """
+        super().__init__()
+
```
Comment (Wovchena) [inline demos/text_to_speech_demo/python/models/mel2wave_ie.py:194]:
```suggestion
```

**Review by Wovchena — APPROVED** [review]

File: demos/text_to_speech_demo/python/text_to_speech_demo.py [hunk 2]
```diff
@@ -49,39 +53,55 @@ def build_argparser():
...
-                      required=False,
-                      type=int)
-
     args.add_argument("-d", "--device",
                       help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL, MYRIAD or HETERO is "
                            "acceptable. The sample will look for a suitable plugin for device specified. "
                            "Default value is CPU",
                       default="CPU", type=str)
+
+    vocoder_parsers = parser.add_subparsers(dest='vocoder')
```
Comment (IRDonch) [inline demos/text_to_speech_demo/python/text_to_speech_demo.py:67]: Subcommands don't seem like the best choice for choosing the model architecture. For one, purely semantically the vocoder type isn't a subcommand. But also, consider what'll happen if in the future you need to add a similar selection for the mel-spectrogram generator. You can't specify two subcommands.

IMO, you should just add all possible model options to the main parser and then make sure that incompatible options aren't specified. `add_mutually_exclusive_group` can help with this.

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: demos/text_to_speech_demo/python/text_to_speech_demo.py [hunk 3]
```diff
@@ -26,12 +26,16 @@
...
 from models.forward_tacotron_ie import ForwardTacotronIE
-from models.mel2wave_ie import WaveRNNIE
+from models.mel2wave_ie import WaveRNNIE, MelGANIE
 
 
-def save_wav(x, path):
+def save_wav(x, path, is_wavernn):
     sr = 22050
-    audio = (x * (2 ** 15 - 1)).astype("<h")
+    if is_wavernn:
```
[the remark this answers is not in the document]
Comment (a-a-egorovich) [reply demos/text_to_speech_demo/python/text_to_speech_demo.py]: Done

File: demos/text_to_speech_demo/python/text_to_speech_demo.py [hunk 4]
```diff
@@ -117,9 +137,10 @@ def main():
                 time_e = time.perf_counter()
                 time_forward += (time_e - time_s) * 1000
 
-                mel = (mel + 4) / 8
-                np.clip(mel, 0, 1, out=mel)
-                mel = np.transpose(mel)
+                if is_wavernn:
+                    mel = (mel + 4) / 8
+                    np.clip(mel, 0, 1, out=mel)
+                    mel = np.transpose(mel)
```
[the remark this answers is not in the document]
Comment (a-a-egorovich) [reply demos/text_to_speech_demo/python/text_to_speech_demo.py]: Done

File: demos/text_to_speech_demo/python/text_to_speech_demo.py (hunk 2 above)
Comment (a-a-egorovich) [reply demos/text_to_speech_demo/python/text_to_speech_demo.py]: Done. As I understood from documentation add_mutually_exclusive_group is not good choice for groups like [m_melgan | m_rnn & m_upsample]. It is possible only split by  [m_melgan | m_rnn | m_upsample]

File: demos/text_to_speech_demo/python/text_to_speech_demo.py [hunk 5]
```diff
@@ -49,43 +49,64 @@ def build_argparser():
...
+        print('Can not use m_rnn and m_upsample with m_melgan. Define m_melgan or [m_rnn, m_upsample]')
+        return False
+    return True
+
+
 def main():
     args = build_argparser().parse_args()
 
+    if not is_correct_args(args):
+        return 0
```
Comment (IRDonch) [inline demos/text_to_speech_demo/python/text_to_speech_demo.py:96]:
```suggestion
        return 1
```

**Comment by IRDonch** [comment]
Please rebase on top of latest `develop`, so that we don't have duplicate `unidecode` removal in the history.

File: demos/text_to_speech_demo/python/text_to_speech_demo.py (hunk 2 above)
Comment (IRDonch) [reply demos/text_to_speech_demo/python/text_to_speech_demo.py]: What I meant was that you could make, for example, `m_melgan` and `m_rnn` mutually exclusive, and then check the presence of the other options based on which one was specified. The current approach is fine too, though.

File: demos/text_to_speech_demo/python/text_to_speech_demo.py (hunk 5 above)
Comment (a-a-egorovich) [reply demos/text_to_speech_demo/python/text_to_speech_demo.py]: Done

**Review by IRDonch — APPROVED** [review]

**Pull request merged**